File: AdminPanel/AdminPanel.Web/Modules/Conne/Cart/CartForm.cs
using Serenity.ComponentModel;

namespace AdminPanel.Conne.Forms;

[FormScript("Conne.Cart")]
[BasedOnRow(typeof(CartRow), CheckNames = true)]
public class CartForm
{
    public string Userid { get; set; }
    public int Productid { get; set; }
    public int Count { get; set; }
    public int Price { get; set; }
}

File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/Conne/CartForm.ts
import { StringEditor, ServiceLookupEditor, IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface CartForm {
    Userid: StringEditor;
    Productid: ServiceLookupEditor;
    Count: IntegerEditor;
    Price: IntegerEditor;
}

export class CartForm extends PrefixedContext {
    static readonly formKey = 'Conne.Cart';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CartForm.init)  {
            CartForm.init = true;

            var w0 = StringEditor;
            var w1 = ServiceLookupEditor;
            var w2 = IntegerEditor;

            initFormType(CartForm, [
                'Userid', w0,
                'Productid', w1,
                'Count', w2,
                'Price', w2
            ]);
        }
    }
}